### PR TITLE
add "enable_fxaa" in rendering.global.json to enable/disable fxaa pass

### PR DESCRIPTION
**Pull request by jiangdunchun**

### CODE REVIEW

File: engine/source/runtime/resource/res_type/global/global_rendering.h [hunk 1]
```diff
@@ -52,6 +52,7 @@ namespace Pilot
         REFLECTION_BODY(GlobalRenderingRes);
 
     public:
+        bool                m_enable_fxaa;
```
Comment (hyv1001) [inline engine/source/runtime/resource/res_type/global/global_rendering.h:55]: need a default variable initialization `bool m_enable_fxaa {true};`

File: engine/source/runtime/function/render/render_pipeline_base.h [hunk 2]
```diff
@@ -14,6 +14,7 @@ namespace Pilot
 
     struct RenderPipelineInitInfo
     {
+        bool enable_fxaa;
```
Comment (hyv1001) [inline engine/source/runtime/function/render/render_pipeline_base.h:17]: need a default variable initialization bool enable_fxaa {false};

File: engine/source/runtime/function/render/render_pipeline.cpp [hunk 3]
```diff
@@ -192,8 +194,8 @@ namespace Pilot
         fxaa_pass.updateAfterFramebufferRecreate(
             main_camera_pass.getFramebufferImageViews()[_main_camera_pass_post_process_buffer_odd]);
         combine_ui_pass.updateAfterFramebufferRecreate(
-            main_camera_pass.getFramebufferImageViews()[_main_camera_pass_backup_buffer_even],
-            main_camera_pass.getFramebufferImageViews()[_main_camera_pass_backup_buffer_odd]);
+            main_camera_pass.getFramebufferImageViews()[_main_camera_pass_backup_buffer_odd],
```
Comment (hyv1001) [inline engine/source/runtime/function/render/render_pipeline.cpp:197]: why change the order?

File: engine/source/runtime/function/render/passes/main_camera_pass.cpp [hunk 4]
```diff
@@ -27,6 +27,9 @@ namespace Pilot
     {
         RenderPass::initialize(nullptr);
 
+        const MainCameraPassInitInfo* _init_info = static_cast<const MainCameraPassInitInfo*>(init_info);
```
Comment (hyv1001) [inline engine/source/runtime/function/render/passes/main_camera_pass.cpp:30]: local variable naming should be `const MainCameraPassInitInfo* init_info`

**Review by hyv1001 — CHANGES_REQUESTED** [review]

File: engine/source/runtime/function/render/render_pipeline.cpp (hunk 3 above)
Comment (jiangdunchun) [reply engine/source/runtime/function/render/render_pipeline.cpp:197]: the `_main_camera_pass_backup_buffer_odd` VkImageView is used as the `scene_input_attachment` now no matter enable or disable FXAA pass. and the `_main_camera_pass_backup_buffer_even` is used as the `ui_input_attachment`. it doesn't make any difference if not changed the order. for easy understanding, i think the parameter should follow the definition `void CombineUIPass::updateAfterFramebufferRecreate(VkImageView scene_input_attachment, VkImageView ui_input_attachment)`.

File: engine/source/runtime/function/render/passes/main_camera_pass.cpp (hunk 4 above)
Comment (jiangdunchun) [reply engine/source/runtime/function/render/passes/main_camera_pass.cpp:30]: the `init_info`  has been used in `const RenderPassInitInfo* init_info` as the parameter of `initialize` function. i referenced the  `initialize` function of `ColorGradingPass` which used this naming style.

File: engine/source/runtime/function/render/render_pipeline_base.h (hunk 2 above)
Comment (jiangdunchun) [reply engine/source/runtime/function/render/render_pipeline_base.h:17]: well, i will set the default variable, thanks

**Review by kwbm — APPROVED** [review]

**Comment by jiangdunchun** [comment]
@hyv1001 please check and assess if these changes should be accepted

**Pull request merged**